### ISSUE TITLE
Possible problem that some formulas in Logits Alignment do not correspond to codes

### ISSUE THREAD

**Issue by WeiLeZhang** [G-U-N/ECCV22-FOSTER#10]
Thank you for such a good work, I find that possible problem about Logits Alignment. The following is a detailed description.

Equation (13) and (14) from the paper:
![image](https://user-images.githubusercontent.com/103939997/206209670-193681bd-b5ac-4053-9c3f-ee21f63aa9d1.png)
Implemented code from `foster.py`：
![image](https://user-images.githubusercontent.com/103939997/206210631-26c97129-7e14-4337-81c0-6fe582187674.png)
![image](https://user-images.githubusercontent.com/103939997/206210865-a374b1b5-af0a-4a73-90eb-8e88b212aca1.png)
According to the code, I calculated beta1 and beta2 as follows:
![image](https://user-images.githubusercontent.com/103939997/206216974-515892c5-f3c2-4461-8585-850c0d034f88.png)
I guess that this result is not equivalent to what is proposed in the paper.

I'm looking forward to your answer!

**Comment by G-U-N** [comment]
Hi, thanks for pointing this out.

The actual formula for computing the $\gamma_1$ and $\gamma_2$ in the code is $\gamma_1 = \frac{c_{old}E_{new} + c_{new}E_{old}}{(c_{old}+c_{new})E_{new}} \le 1$, $\gamma_2 = \frac{c_{old}E_{new} + c_{new}E_{old}}{(c_{old}+c_{new})E_{old}} \ge 1$, where $c_{old}$, and $c_{new}$ means the number of old and new classes, respectively.   In this way, we could force the model to enlarge the absolute value of logits on old categories and reduce that on new categories. 

We express the slightly simplified and different formula in the paper to  make our idea more straightforward and acceptable. (since it still follows that $\gamma_1 \le  \gamma_2$.) 

Hope this clarifies your queries.

**Comment by WeiLeZhang** [comment]
Hi, I get it. Thanks for answering my question quickly!
